core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
-    if ! kubectl delete secret -n ${NAMESPACE} kcpsec ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete secret -n ${NAMESPACE} kcpsec ; then
         echo "Nothing to delete."
     fi
-    kubectl get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > kcpsecret
-    kubectl create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig="kcpsecret"    
-    rm kcpsecret
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig=$kcp_kubeconfig   
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete spaceproviderdesc $PROVIDER_NAME ; then
         echo "Nothing to delete."
     fi
-    kubectl create -f - <<EOF
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+    echo "Waiting for spaceprovider to reach the Ready phase."
+    kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} wait --for=jsonpath='{.status.Phase}'=Ready spaceproviderdesc $PROVIDER_NAME
+}
  
-        echo "Delete the kubeflex provider object if it already exists, and then create it."
-        if ! kubectl delete spaceproviderdesc pkflex ; then
-            echo "Nothing to delete."
-        fi
-        kubectl create -f - <<EOF
+function create_kubeflex_provider() {
+    echo "Waiting for the kubeflex provider to be ready... this may take a while."
+    (
+        until [ "$(kubectl --kubeconfig $kubestellar_kubeconfig get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
+            sleep 10
+        done
+    )
+
+    echo "Delete the kubeflex provider object if it already exists, and then create it."
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete spaceproviderdesc $PROVIDER_NAME ; then
+        echo "Nothing to delete."
+    fi
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkflex
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kubeflex"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: corecluster
 EOF
-    fi
-
-    KUBECONFIG=$oldKubeconfig
+    echo "Waiting for spaceprovider to reach the Ready phase."
```

Comment:
This function is printing progress messages about the provider (KubeFlex) and about the SpaceProviderDesc. You do not have the luxury here to claim that they are synonymous. Better here to say exactly what is meant ("SpaceProviderDesc", rather than "spaceprovider").

Reply:
Why was this marked "resolved"?

Reply:
I changes the text to SpaceProviderDesc.  Is this sufficient?